### PR TITLE
docs: Fix some docstring errors in torch.nn.utils parametrize/spectral_norm/stateless

### LINKED ISSUE #112630: Fix docstring errors in parametrize.py, spectral_norm.py, stateless.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `cached`, **Line**: 21, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `cached`, **Line**: 21, **Description**: First line should end with a period (not 's')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `ParametrizationList`, **Line**: 70, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `ParametrizationList`, **Line**: 70, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `right_inverse`, **Line**: 201, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `right_inverse`, **Line**: 201, **Description**: First line should end with a period (not ')')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `right_inverse`, **Line**: 201, **Description**: First line should be in imperative mood (perhaps 'Call', not 'Calls')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `_inject_new_class`, **Line**: 280, **Description**: First line should be in imperative mood (perhaps 'Set', not 'Sets')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `register_parametrization`, **Line**: 381, **Description**: First line should be in imperative mood (perhaps 'Add', not 'Adds')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `is_parametrized`, **Line**: 587, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `remove_parametrizations`, **Line**: 609, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `remove_parametrizations`, **Line**: 609, **Description**: First line should be in imperative mood (perhaps 'Remove', not 'Removes')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `type_before_parametrizations`, **Line**: 690, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `type_before_parametrizations`, **Line**: 690, **Description**: First line should end with a period (not ',')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `type_before_parametrizations`, **Line**: 690, **Description**: First line should be in imperative mood (perhaps 'Return', not 'Returns')
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `transfer_parametrizations_and_params`, **Line**: 704, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/utils/parametrize.py`, **Entity**: `transfer_parametrizations_and_params`, **Line**: 704, **Description**: First line should end with a period (not 'e')
- **File**: `torch/nn/utils/spectral_norm.py`, **Entity**: ``, **Line**: 1, **Description**: One-line docstring should fit on one line with quotes (found 3)
- **File**: `torch/nn/utils/spectral_norm.py`, **Entity**: ``, **Line**: 1, **Description**: First line should end with a period (not '7')
- **File**: `torch/nn/utils/spectral_norm.py`, **Entity**: `spectral_norm`, **Line**: 226, **Description**: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
- **File**: `torch/nn/utils/spectral_norm.py`, **Entity**: `remove_spectral_norm`, **Line**: 287, **Description**: First line should be in imperative mood (perhaps 'Remove', not 'Removes')
- **File**: `torch/nn/utils/stateless.py`, **Entity**: `functional_call`, **Line**: 156, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/utils/stateless.py`, **Entity**: `functional_call`, **Line**: 156, **Description**: First line should end with a period (not 's')
- **File**: `torch/nn/utils/stateless.py`, **Entity**: `functional_call`, **Line**: 156, **Description**: First line should be in imperative mood (perhaps 'Perform', not 'Performs')

### PR DESCRIPTION
**Pull request by bjhargrave**
Fixes https://github.com/pytorch/pytorch/issues/112630

cc @svekars @carljparker

### CODE REVIEW

**Comment by bjhargrave** [comment]
BEFORE: 
```
(pytorch-dev) ➜  pytorch git:(main) ✗ pydocstyle --count torch/nn/utils/parametrize.py
torch/nn/utils/parametrize.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/utils/parametrize.py:21 in public function `cached`:
        D205: 1 blank line required between summary line and description (found 0)
torch/nn/utils/parametrize.py:21 in public function `cached`:
        D400: First line should end with a period (not 's')
torch/nn/utils/parametrize.py:70 in public class `ParametrizationList`:
        D204: 1 blank line required after class docstring (found 0)
torch/nn/utils/parametrize.py:70 in public class `ParametrizationList`:
        D205: 1 blank line required between summary line and description (found 0)
torch/nn/utils/parametrize.py:97 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/parametrize.py:201 in public method `right_inverse`:
        D205: 1 blank line required between summary line and description (found 0)
torch/nn/utils/parametrize.py:201 in public method `right_inverse`:
        D400: First line should end with a period (not ')')
torch/nn/utils/parametrize.py:201 in public method `right_inverse`:
        D401: First line should be in imperative mood (perhaps 'Call', not 'Calls')
torch/nn/utils/parametrize.py:261 in public method `forward`:
        D102: Missing docstring in public method
torch/nn/utils/parametrize.py:280 in private function `_inject_new_class`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
torch/nn/utils/parametrize.py:381 in public function `register_parametrization`:
        D401: First line should be in imperative mood (perhaps 'Add', not 'Adds')
torch/nn/utils/parametrize.py:587 in public function `is_parametrized`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
torch/nn/utils/parametrize.py:609 in public function `remove_parametrizations`:
        D202: No blank lines allowed after function docstring (found 1)
torch/nn/utils/parametrize.py:609 in public function `remove_parametrizations`:
        D401: First line should be in imperative mood (perhaps 'Remove', not 'Removes')
torch/nn/utils/parametrize.py:690 in public function `type_before_parametrizations`:
        D205: 1 blank line required between summary line and description (found 0)
torch/nn/utils/parametrize.py:690 in public function `type_before_parametrizations`:
        D400: First line should end with a period (not ',')
torch/nn/utils/parametrize.py:690 in public function `type_before_parametrizations`:
        D401: First line should be in imperative mood (perhaps 'Return', not 'Returns')
torch/nn/utils/parametrize.py:704 in public function `transfer_parametrizations_and_params`:
        D205: 1 blank line required between summary line and description (found 0)
torch/nn/utils/parametrize.py:704 in public function `transfer_parametrizations_and_params`:
        D400: First line should end with a period (not 'e')
20
(pytorch-dev) ➜  pytorch git:(main) ✗ pydocstyle --count torch/nn/utils/spectral_norm.py 
torch/nn/utils/spectral_norm.py:1 at module level:
        D200: One-line docstring should fit on one line with quotes (found 3)
torch/nn/utils/spectral_norm.py:1 at module level:
        D400: First line should end with a period (not '7')
torch/nn/utils/spectral_norm.py:12 in public class `SpectralNorm`:
        D101: Missing docstring in public class
torch/nn/utils/spectral_norm.py:27 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/spectral_norm.py:36 in public method `reshape_weight_to_matrix`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:45 in public method `compute_weight`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:97 in public method `remove`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:106 in public method `__call__`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:117 in public method `apply`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:158 in public class `SpectralNormLoadStateDictPreHook`:
        D101: Missing docstring in public class
torch/nn/utils/spectral_norm.py:160 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/spectral_norm.py:171 in public method `__call__`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:204 in public class `SpectralNormStateDictHook`:
        D101: Missing docstring in public class
torch/nn/utils/spectral_norm.py:206 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/spectral_norm.py:209 in public method `__call__`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:225 in public function `spectral_norm`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
torch/nn/utils/spectral_norm.py:286 in public function `remove_spectral_norm`:
        D401: First line should be in imperative mood (perhaps 'Remove', not 'Removes')
17
(pytorch-dev) ➜  pytorch git:(main) ✗ pydocstyle --count torch/nn/utils/stateless.py     
torch/nn/utils/stateless.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/utils/stateless.py:154 in public function `functional_call`:
        D205: 1 blank line required between summary line and description (found 0)
torch/nn/utils/stateless.py:154 in public function `functional_call`:
        D400: First line should end with a period (not 's')
torch/nn/utils/stateless.py:154 in public function `functional_call`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
4
```

AFTER:
```
(pytorch-dev) ➜  pytorch git:(main) ✗ pydocstyle --count torch/nn/utils/parametrize.py
torch/nn/utils/parametrize.py:1 at module level:
        D100: Missing docstring in public module
torch/nn/utils/parametrize.py:96 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/parametrize.py:260 in public method `forward`:
        D102: Missing docstring in public method
3
(pytorch-dev) ➜  pytorch git:(main) ✗ pydocstyle --count torch/nn/utils/spectral_norm.py
torch/nn/utils/spectral_norm.py:10 in public class `SpectralNorm`:
        D101: Missing docstring in public class
torch/nn/utils/spectral_norm.py:25 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/spectral_norm.py:34 in public method `reshape_weight_to_matrix`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:43 in public method `compute_weight`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:95 in public method `remove`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:104 in public method `__call__`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:115 in public method `apply`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:156 in public class `SpectralNormLoadStateDictPreHook`:
        D101: Missing docstring in public class
torch/nn/utils/spectral_norm.py:158 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/spectral_norm.py:169 in public method `__call__`:
        D102: Missing docstring in public method
torch/nn/utils/spectral_norm.py:202 in public class `SpectralNormStateDictHook`:
        D101: Missing docstring in public class
torch/nn/utils/spectral_norm.py:204 in public method `__init__`:
        D107: Missing docstring in __init__
torch/nn/utils/spectral_norm.py:207 in public method `__call__`:
        D102: Missing docstring in public method
13
(pytorch-dev) ➜  pytorch git:(main) ✗ pydocstyle --count torch/nn/utils/stateless.py
torch/nn/utils/stateless.py:1 at module level:
        D100: Missing docstring in public module
1
```

File: torch/nn/utils/parametrize.py [hunk 1]
```diff
@@ -277,7 +276,7 @@ def forward(self) -> Tensor:
 
 
 def _inject_new_class(module: Module) -> None:
-    r"""Sets up a module to be parametrized.
+    r"""Set up a module to be parametrized.
```
Comment (lezcano) [inline torch/nn/utils/parametrize.py:279]: Why this change? Same for the equivalent one in right_inverse and many others below.

These sentences refer to the function at hand, so they should be in third person. This is how it's done in most of the docs.

File: torch/nn/utils/parametrize.py [hunk 2]
```diff
@@ -584,15 +583,16 @@ def right_inverse(self, X: Tensor) -> Union[Tensor, Sequence[Tensor]]
...
+    r"""Determine if a module has a parametrization.
 
     Args:
         module (nn.Module): module to query
-        tensor_name (str, optional): attribute in the module to query
+        tensor_name (str, optional): name of the parameter in the module
             Default: ``None``
+    Returns:
+        ``True`` if :attr:`module` has any parametrization and :attr:`tensor_name` is ``None``;
+        ``True`` if :attr:`module` has a parametrization for the parameter named :attr:`tensor_name`;
```
Comment (lezcano) [inline torch/nn/utils/parametrize.py:594]: What about
```suggestion
        ``True`` if :attr:`module` has a parametrization for the parameter named :attr:`tensor_name`, or if it has any parametrization, if :attr:`tensor_name` is ``None``;
```

File: torch/nn/utils/parametrize.py [hunk 3]
```diff
@@ -701,8 +699,9 @@ def type_before_parametrizations(module: Module) -> type:
 def transfer_parametrizations_and_params(
     from_module: Module, to_module: Module, tensor_name: Optional[str] = None
 ) -> Module:
-    r"""Transfers parametrizations and the parameters they parametrize from from_module
-    to to_module. If tensor_name is specified, only transfers the specified parameter, otherwise
+    r"""Transfer parametrizations and the parameters they parametrize from from_module to to_module.
```
Comment (lezcano) [inline torch/nn/utils/parametrize.py:702]: Preexisting, but from_module and to_module, and tensor_name in the next line should be `:attr:`

**Review by lezcano — COMMENTED** [review]
The rewrites are great! I have a few points in a few of them tho.

File: torch/nn/utils/parametrize.py (hunk 1 above)
Comment (svekars) [reply torch/nn/utils/parametrize.py:279]: @lezcano this is the change needed by pydocstyle - we are working on standardizing docstrings for pydocstyle.

Comment (lezcano) [reply torch/nn/utils/parametrize.py:279]: Fair enough. I personally prefer docs as a description of the function, but I guess that as long as it's consistent, the rest is less important.

**Comment by kit1980** [comment]
@bjhargrave please give more descriptive title to the PR to distinguish it from others, maybe mention path.

File: torch/nn/utils/parametrize.py (hunk 1 above)
Comment (bjhargrave) [reply torch/nn/utils/parametrize.py:279]: This addresses the `D401: First line should be in imperative mood` [pydocstyle rule](https://www.pydocstyle.org/en/stable/error_codes.html) which is in the list of fixes we were asked to address in the docathon issue.

```
torch/nn/utils/parametrize.py:280 in private function `_inject_new_class`:
        D401: First line should be in imperative mood (perhaps 'Set', not 'Sets')
```

**Review by lezcano — APPROVED** [review]
Thank you for the clean-up and the great rewrites!
